From ffb8ddb91bbdbbce1604de6088e456f46e6618ff Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Tue, 28 Nov 2017 13:19:10 -0500
Subject: [PATCH 115/116] prueth: improve debugfs clean up code using existing
 API

Use existing API, debugfs_remove_recursive(), to simplify clean up
code related to debugfs.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/prueth.c | 20 ++++----------------
 1 file changed, 4 insertions(+), 16 deletions(-)

diff --git a/drivers/net/ethernet/ti/prueth.c b/drivers/net/ethernet/ti/prueth.c
index 1dd96be..a69a4e3 100644
--- a/drivers/net/ethernet/ti/prueth.c
+++ b/drivers/net/ethernet/ti/prueth.c
@@ -1277,9 +1277,8 @@ int prueth_hsr_prp_debugfs_init(struct prueth *prueth)
 void
 prueth_debugfs_term(struct prueth_emac *emac)
 {
-	debugfs_remove(emac->stats_file);
+	debugfs_remove_recursive(emac->root_dir);
 	emac->stats_file = NULL;
-	debugfs_remove(emac->root_dir);
 	emac->root_dir = NULL;
 }
 
@@ -1299,25 +1298,14 @@ int prueth_hsr_prp_debugfs_init(struct prueth *prueth)
 	if (prueth->emac_configured)
 		return;
 
-	debugfs_remove(prueth->node_tbl_file);
+	debugfs_remove_recursive(prueth->root_dir);
 	prueth->node_tbl_file = NULL;
-	debugfs_remove(prueth->nt_clear_file);
 	prueth->nt_clear_file = NULL;
-	if (PRUETH_IS_HSR(prueth)) {
-		debugfs_remove(prueth->hsr_mode_file);
-		prueth->hsr_mode_file = NULL;
-	}
-	debugfs_remove(prueth->dlrmt_file);
+	prueth->hsr_mode_file = NULL;
 	prueth->dlrmt_file = NULL;
-	debugfs_remove(prueth->dd_file);
 	prueth->dd_file = NULL;
-	if (PRUETH_IS_PRP(prueth)) {
-		debugfs_remove(prueth->tr_file);
-		prueth->tr_file = NULL;
-	}
-	debugfs_remove(prueth->error_stats_file);
+	prueth->tr_file = NULL;
 	prueth->error_stats_file = NULL;
-	debugfs_remove(prueth->root_dir);
 	prueth->root_dir = NULL;
 }
 #else
-- 
1.9.1

